base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -667,70 +704,76 @@ impl LMDBDatabase {
             &(header.height, header.hash().as_slice()),
             "output_mmr_size_index",
         )?;
-        Ok(true)
+        Ok(())
     }
 
-    fn op_delete(&mut self, txn: &WriteTransaction<'_>, key: DbKey) -> Result<(), ChainStorageError> {
-        match key {
-            DbKey::BlockHeader(k) => {
-                let val: Option<BlockHeader> = lmdb_get(txn, &self.headers_db, &k)?;
-                if let Some(v) = val {
-                    let hash = v.hash();
-                    // Check that there are no utxos or kernels linked to this.
+    fn delete_header(&mut self, txn: &WriteTransaction<'_>, height: u64) -> Result<(), ChainStorageError> {
```

Comment:
```suggestion
    fn delete_header(&mut self, txn: &WriteTransaction<'_>, hash: Vec<u8>) -> Result<(), ChainStorageError> {
```

This will make sure we delete the exact header we want, and not the header at the exact height